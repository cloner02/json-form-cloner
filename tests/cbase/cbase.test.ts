const  { CBase, CForm} = require('json-form-cloner');
describe('CBase component', () => {
    const CBASE_TAG = 'c-base';
    const getShadowRoot = (tagName: string): ShadowRoot | null => {
        return document.body.getElementsByTagName(tagName)[0].shadowRoot;
    };
    it('renders correctly', () => {
        const cBase = new CBase();

        expect(cBase.render()).toMatchSnapshot();
    });
    it('dom renders correctly with the value property', async () => {
        const cBase = new CBase();
        cBase.value = 'hello world';
        window.document.body.appendChild(cBase);
        await cBase.updateComplete;
        const renderedCbase = getShadowRoot(CBASE_TAG)?.innerHTML;
        expect(renderedCbase?.indexOf(cBase.value)).not.toBe(-1);
    });
});
