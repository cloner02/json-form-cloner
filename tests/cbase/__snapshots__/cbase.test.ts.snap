// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CBase component renders correctly 1`] = `undefined`;
